atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/AdjustableSweepBatchConfigSourceTest.java

```diff
@@ -0,0 +1,216 @@
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+/**
+ * Created by tboam on 03/11/2017.
+ */
+public class AdjustableSweepBatchConfigSourceTest {
+
+    private AdjustableSweepBatchConfigSource adjustableConfig;
+    private double previousMultiplier;
+    private int previousMaxCellTsPairsToExamine;
+    private int previousCandidateBatchSize;
+    private int previousDeleteBatchSize;
+
+    @Test
+    public void batchSizeMultiplierDecreasesOnFailure() {
+        // Given
+        configWithValues(1000, 1000, 1000);
+
+        // When
+        adjustableConfig.decreaseMultiplier();
+
+        // Then
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(lessThan(previousMultiplier)));
+    }
+
+    @Test
+    public void canDecreaseAndIncreaseConfigWithAllSmallValues() {
+        //Given
+        configWithValues(1, 1, 1);
+
+        whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease();
+
+        whenIncreasingTheMultiplier_thenAdjustedConfigValuesIncrease();
+
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(1.0));
+    }
+
+    @Test
+    public void canDecreaseAndIncreaseConfigWithAllLargeValues() {
+        //Given
+        configWithValues(1000, 1000, 1000);
+
+        whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease();
+
+        whenIncreasingTheMultiplier_thenAdjustedConfigValuesIncrease();
+
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(1.0));
+    }
+
+    @Test
+    public void canDecreaseAndIncreaseConfigWithMixOfValues() {
+        //Given
+        configWithValues(1000, 1, 100);
+
+        whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease();
+
+        whenIncreasingTheMultiplier_thenAdjustedConfigValuesIncrease();
+
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(1.0));
+    }
+
+    private void whenIncreasingTheMultiplier_thenAdjustedConfigValuesIncrease() {
+        for (int i = 0; i < 1_000; i++) {
+            // When
+            adjustableConfig.increaseMultiplier();
+
+            // Then
+            batchSizeMultiplierIncreases();
+            batchSizeMultiplierDoesNotExceedOne();
+
+            maxCellTsPairsToExamineDoesNotDecrease();
+            candidateBatchSizeDoesNotDecrease();
+            deleteBatchSizeDoesNotDecrease();
+
+            maxCellTsPairsToExamineDoesNotExceedBaseConfig();
+            candidateBatchSizeDoesNotExceedBaseConfig();
+            deleteBatchSizeDoesNotExceedBaseConfig();
+
+            updatePreviousValues();
+        }
+    }
+
+    private void whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease() {
+        for (int i = 0; i < 10_000; i++) {
+            // When
+            adjustableConfig.decreaseMultiplier();
+
+            // Then
+            batchSizeMultiplierDecreases();
+            maxCellTsPairsToExamineDecreasesToAMinimumOfOne();
```

Comment:
nit coding-style: These three asserts are repetitive, we might be able to colease this into one function and/or use lambdas.

Reply:
I've applied some lambdas to the assertion methods to tidy them up and reduce duplication